providers/azure/resources/redis_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/redis/armredis/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// mockAzureConnection implements the AzureConnection interface for testing
+type mockAzureConnection struct {
+	subscriptionID string
+	token          string
+}
+
+func (m *mockAzureConnection) ID() uint32 {
+	return 1
+}
+
+func (m *mockAzureConnection) Name() string {
+	return "mock-azure-connection"
+}
+
+func (m *mockAzureConnection) SubId() string {
+	return m.subscriptionID
+}
+
+func (m *mockAzureConnection) Token() string {
+	return m.token
+}
+
+func (m *mockAzureConnection) ClientOptions() interface{} {
+	return nil
+}
+
+func (m *mockAzureConnection) ParentID() uint32 {
+	return 0
+}
+
+// MockRedisClientFactory mocks the Azure Redis client factory
+type mockRedisClientFactory struct {
+	mock.Mock
+}
+
+func (m *mockRedisClientFactory) NewClient() *armredis.Client {
+	args := m.Called()
+	return args.Get(0).(*armredis.Client)
+}
+
+// MockRedisClient mocks the Azure Redis client
+type mockRedisClient struct {
+	mock.Mock
+}
+
+func (m *mockRedisClient) NewListBySubscriptionPager(options *armredis.ClientListBySubscriptionOptions) *runtime.Pager[armredis.ClientListBySubscriptionResponse] {
+	args := m.Called(options)
+	return args.Get(0).(*runtime.Pager[armredis.ClientListBySubscriptionResponse])
+}
+
+// mockPager mocks the Azure pager
+type mockPager struct {
+	mock.Mock
+	pages []armredis.ClientListBySubscriptionResponse
+}
+
+func (m *mockPager) More() bool {
+	args := m.Called()
+	return args.Bool(0)
+}
+
+func (m *mockPager) NextPage(ctx context.Context) (armredis.ClientListBySubscriptionResponse, error) {
+	args := m.Called(ctx)
+	return args.Get(0).(armredis.ClientListBySubscriptionResponse), args.Error(1)
+}
+
+// Helper functions to create pointers
+func stringPtr(s string) *string {
```

Comment:
Go supports generics, you don't need to write a separate function for every pointer conversion. You can just do:
```go
func ptr[T any](v T) *T {
  return &v
}
```
Then you can call this for any type